Framework/Logger/include/Framework/Logger.h

```diff
@@ -0,0 +1,36 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#ifndef O2_FRAMEWORK_LOGGER_H_
+#define O2_FRAMEWORK_LOGGER_H_
+
+// FIXME: until we actually have fmt widely available we simply print out the
+// format string.
+// If FairLogger is not available, we use fmt::printf directly, with no level.
+#if __has_include(<fairlogger/Logger.h>)
+#include <fairlogger/Logger.h>
+#if __has_include(<fmt/format.h>)
+#include <fmt/format.h>
+#define LOGF(level, ...) LOG(level) << fmt::format(__VA_ARGS__)
+#else
+#define O2_FIRST_ARG(N, ...) N
+#define LOGF(level, ...) LOG(level) << O2_FIRST_ARG(__VA_ARGS__)
+#endif
+#define O2DEBUG(...) LOGF(debug, __VA_ARGS__)
+#define O2INFO(...) LOGF(info, __VA_ARGS__)
+#define O2ERROR(...) LOGF(error, __VA_ARGS__)
+#elif __has_include(<fmt/format.h>)
+#include <fmt/format.h>
+#define LOGF(level, ...) fmt::printf(__VA_ARGS__)
```

Comment:
Should't this be `fmt::print(__VA_ARGS__)`? At least in the test you use the `"{}", "Hello world"` format.
Looking at their docs:
```
fmt::print("Hello, {}!", "world");  // Python-like format string syntax
fmt::printf("Hello, %s!", "world"); // printf format string syntax
```
Not sure if printf also supports the other formatting..

Reply:
We will add fmt also to FairLogger in short term, that will look something like this macro but honoring the levels and so on.

Reply:
@rbx @ktf : The default case is actually `#define LOGF(level, ...) LOG(level) << fmt::format(__VA_ARGS__)`. (This one also honors the levels.) The fmt::printf is only used in case fairlogger is not available.

In general, I agree: I'd use the fmt::print function, if it is more versatile. In particular, we should use a function with the same features as fmt::format.

One other point: Do fmt::format and fmt::print offer also the full syntax of fmt::printf?
I need the printf style for my debug output in kernel code, which I want to redirect to normal printf for GPU code.
If the syntax for fmt::print and fmt::printf are different, I'd suggest we have two macros: LOGF for printf like syntax, and LOGP (or whatever name) for the python like fmt::print syntax.

Finally, when this is included into FairLogger, I'd appreciate very much if there would also be one printf like option.

Reply:
@davidrohr sure, we can provide both print and printf formats. Not sure yet about the macro names on our side, initially I was also thinking of LOGF for FairLogger, but maybe we will pick something else to avoid conflict^^.
I understand that your default does handle the level, that is fine. The reason why I am focusing on this macro is because for the FairLogger implementation of the LOGF-equivalent I would like to avoid using streams completely for performance reasons. It's a bit more work, but nothing crazy i think.

Reply:
@rbx : OK, makes sense. I think we have the stream with the fmt::format only to use theFairLogger LOG(...) macro for the correct log level. When it is handled inside FairLogger, I agree that avoiding the stream makes more sense. You could also have a look at the discussion with @ktf above, about how to support constexpr strings where possible.